src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] array;
+    private int size;
+    private int maxSize;
 
+    public ArrayList() {
+        size = 0;
+        maxSize = DEFAULT_SIZE;
+        array = (T[]) (new Object[maxSize]);
+    }
+
+    public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size >= maxSize) {
+            array = getExpanded();
+        }
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("You can't add an element at that index");
+        } else if (index == size) {
+            array[index] = value;
+        } else {
+            array = insertAndCopy(value, index);
```

Comment:
Copying is part of insert logic. No need to have this word in the method name.
```suggestion
            array = insertElement(value, index);
```